### ISSUE TITLE
[Bug]: Menu not working with in v9 with React 18

### ISSUE THREAD

**Issue by kirill-linnik** [microsoft/fluentui#26031]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
    Memory: 2.74 GB / 15.60 GB
  Browsers:
    Edge: Spartan (44.22621.963.0), Chromium (108.0.1462.54)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/s/cocky-dan-qfkgkn?file=/src/App.tsx

## Actual Behavior
took an example from official samples and it doesn't work for react 18

code:
```
import "./styles.css";
import {
  Button,
  Menu,
  MenuItem,
  MenuList,
  MenuPopover,
  MenuTrigger
} from "@fluentui/react-components";

export default function App() {
  const NestedSubmenusControlled = () => {
    return (
      <Menu>
        <MenuTrigger disableButtonEnhancement>
          <Button>menu</Button>
        </MenuTrigger>

        <MenuPopover>
          <MenuList>
            <MenuItem>New </MenuItem>
            <MenuItem>New Window</MenuItem>
            <MenuItem disabled>Open File</MenuItem>
            <MenuItem>Open Folder</MenuItem>
          </MenuList>
        </MenuPopover>
      </Menu>
    );
  };

  return <NestedSubmenusControlled />;
}
```

## Expected Behavior
menu opens

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by layershifter** [comment]
Related to #25232.
There is #25956 that should fix it.

**Comment by kirill-linnik** [comment]
tried with the latest (9.7.4), still doesn't work

**Comment by ling1726** [comment]
@kirill-linnik the fix was merged after the release of 9.7.4, with the christmas season coming up, we don't expect this change to be released before 2023
